test/e2e/kic/kubernetes/kong_ingress.go

```diff
@@ -0,0 +1,117 @@
+			Install(YamlK8s(namespaceWithSidecarInjection(TestNamespace))).
+			Install(EchoServerK8s("default")).
+			Setup(kubernetes)
+		Expect(err).ToNot(HaveOccurred())
+		err = kubernetes.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+	E2EAfterEach(func() {
+		Expect(kubernetes.DeleteNamespace(ingressNamespace)).To(Succeed())
+	})
+	E2EAfterSuite(func() {
+		Expect(kubernetes.DeleteKuma(kubernetesOps...)).To(Succeed())
+		Expect(kubernetes.DeleteNamespace(TestNamespace)).To(Succeed())
+		Expect(kubernetes.DismissCluster()).To(Succeed())
+	})
+	It("should install kong ingress into default namespace", func() {
+		ingressNamespace = defaultIngressNamespace
+		// given kong ingress
+		output, err := kubernetes.GetKumactlOptions().RunKumactlAndGetOutputV(Verbose, "install", "gateway", "kong")
+		Expect(err).ToNot(HaveOccurred())
+		err = NewClusterSetup().Install(YamlK8s(output)).Setup(kubernetes)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Wait for ingress
+		err = NewClusterSetup().Install(Combine(
+			WaitNumPodsNamespace(ingressNamespace, 1, ingressApp),
+			WaitPodsAvailable(ingressNamespace, ingressApp))).Setup(kubernetes)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Test connection to echo server through ingress "proxy" pod
+		pods, err := k8s.ListPodsE(
+			kubernetes.GetTesting(),
+			kubernetes.GetKubectlOptions(ingressNamespace),
+			metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", ingressApp),
+			},
+		)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pods).To(HaveLen(1))
+		clientPod := &pods[0]
+
+		_, _, err = kubernetes.ExecWithRetries(ingressNamespace, clientPod.GetName(), "proxy",
+			"wget", "-O-", "echo-server_kuma-test_svc_80.mesh")
+		Expect(err).ToNot(HaveOccurred())
+	})
+	It("should install kong ingress into non-default namespace", func() {
+		ingressNamespace = altIngressNamespace
+		// given kong ingress
+		output, err := kubernetes.GetKumactlOptions().RunKumactlAndGetOutputV(Verbose, "install", "gateway", "kong", "--namespace", ingressNamespace)
+		Expect(err).ToNot(HaveOccurred())
+		err = NewClusterSetup().Install(YamlK8s(output)).Setup(kubernetes)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Wait for ingress
+		err = NewClusterSetup().Install(Combine(
+			WaitNumPodsNamespace(ingressNamespace, 1, ingressApp),
+			WaitPodsAvailable(ingressNamespace, ingressApp))).Setup(kubernetes)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Test connection to echo server through ingress "proxy" pod
+		pods, err := k8s.ListPodsE(
+			kubernetes.GetTesting(),
+			kubernetes.GetKubectlOptions(ingressNamespace),
+			metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", ingressApp),
+			},
+		)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pods).To(HaveLen(1))
+		clientPod := &pods[0]
+
+		_, _, err = kubernetes.ExecWithRetries(ingressNamespace, clientPod.GetName(), "proxy",
+			"wget", "-O-", "echo-server_kuma-test_svc_80.mesh")
+		Expect(err).ToNot(HaveOccurred())
+	})
+}
```

Comment:
I like that you made an effort to not recreate the whole environment with every `It()`.

A couple of things
1)  For the test itself. I see that the test is about _deploying_ Kong. Then you send the traffic by exec into the pod and run wget. This way traffic is not sent through Kong.
I'd really love to have a test that checks that traffic is sent through Kong so how about we deploy `Ingress` CRD and try to send the traffic via Kong?
2) I very much see KIC to be reused in other tests. Can we try to contain KIC in `deployments` abstraction (check `tests/framework/deployments`? Check for example how we deploy tracing. IMHO the long-term goal is that everything we deploy on the cluster is a `deployment` including Kuma itself. For Kong, I'd love to do `NewClusterSetup().Install(KongIngressController(WithNamespace("xyz"))).Setup(kubernetes)`

Reply:
OK, built KongIngress into framework to support `Install(KongIngressController(WithNamespace("xyz")))`, added e2e framework interfaces supporting Ingress, and now connect to echo server via kong ingress to verify successful KIC setup.